### DISCUSSION TITLE
Issues with monitoring battery voltage on bigquad deck

### DISCUSSION THREAD

**Discussion by simonhc24** [bitcraze/discussions#194] [Q&A]
Hello, 

I am currently building a drone using the BigQuad Deck, and an 11.1 V battery.  

The motors are powered through a separate power distribution board, and I am running the Crazyflie and Big Quad Deck through a 5v power regulator. 

I need to monitor the current battery voltage on the crazyflie, but I can't hook up the battery to the crazyflie otherwise I will burn it out. 

From what I understand the Big Quad deck should have external voltage monitoring capabilities, and I have the DBQ_DECK_ENABLE_PM flag in my make folder, and connected one of the pdb terminals to the Mon port on the BQD. However, I am unsure how to get a voltage with this enabled. 

The vbat value randomly goes between ~4v and 0.11v, an issue that seems to be common to [this](https://github.com/orgs/bitcraze/discussions/98) discussion, while extVbat remains constantly at 0. 

Any help or suggestions on getting an accurate battery reading would be greatly appreciated

**Comment by knmcguire** [comment]
Hi!

@tobbeanton would be the best to answer this and he'll be back next week from his holidays. I'll ping him then to help out.

**Reply by simonhc24** [reply]
Hello again!

This may be a separate discussion thread, but I tried to monitor the voltage by putting the battery through a voltage divider and running the output of that into an analog port on the crazyflie, to no success. So I was also wondering if you had any suggestions or advice on that. 

I was also wondering if you had any updates on my initial question. 

Thank you!

**Reply by knmcguire** [reply]
Hi!

The new question you should make a new discussion for

The initial question will be answered in the next few days

**Reply by tobbeanton** [reply]
The BigQuad functionality is unfortunately not fully implemented. I've checked the voltage measurement and it works but is not reported to the client battery bar. It is instead in pm.extVbat or pm.extVbatMV log variables. We should add a configuration option so that it is possible to overwrite vbat if one wants to. I've created issue [#1096](https://github.com/bitcraze/crazyflie-firmware/issues/1096) to track this.
